terraso_backend/apps/core/models/users.py

```diff
@@ -81,3 +83,19 @@ def is_group_manager(self, group_id):
 
     def __str__(self):
         return self.email
+
+
+class UserPreference(BaseModel):
+    key = models.CharField(max_length=128)
+    value = models.TextField(max_length=512, blank=True, default="")
```

Comment:
Suggestion: checking the structure now of preferences, what do you think of adding a `JSONField` on `User` model? We have a field like that for polygon: https://github.com/techmatters/terraso-backend/blob/67f6ee452d4bd2af51a64890da3a5b6f01eb27a4/terraso_backend/apps/core/models/landscapes.py#L30

Reply:
Just for context, we have good conditions to make queries on `JSONField`: https://docs.djangoproject.com/en/4.0/topics/db/queries/#querying-jsonfield

Reply:
@cacarraraThat is a good suggestion, as long as we are able to just query the specific preferences the UI needs, but I think that will require custom graphQL resolvers, right? or else it will be just one `preferences`field in the UserNode returning everything. So from the `models` side it will jsut be one extra field in the User model called preferences, but in the GraphQL schema it will have the same pieces, what do you think?

Reply:
@cacarrara I do see some cases:
- Preferences related to lists pagination
- Preferences related to map size display
- Prefrences related to notifications
- Preferences related to emails
Those ideally should be fetched in the section or module where they are needed, but I do understand those are not requirements right now, but likely will be.

I think in general I have more concerns related to keeping the structure of the preferences field not controlled. For example: 
- Eventually we will want to generate emails from a different service and that service will require to know the locale selected by the user, since the api is not saying anything related to the structure then this will be a coordination between this new module or service and the web client related to the structure of the preferences field.
- Also, if a different service needs to set user preferences they will have a similar issue, they will have to coordinate at some level with the rest of the services to know what the complete structure is instead of just the key of the preference.

Another downside with the approach (if eventually we identify it is not enough) it will require data schema changes and data migration that are in general expensive and complex to coordinate, so a simplistic approach right now might produce a difficult to revert state.

So, I think we have some options:
1. We can keep preferences related to each service, like web client preferences, mailing service preferences, api preferences, etc.. but this will not let us reuse preference between services
2. Keep the current implementation and set the `value` field in `UserPreference` to be a `JSONField` this will give us more structure inside that value.
3. Change it to just one field in the User model but keep the structure from the GraphQL layer

What do you think?

Reply:
Let's keep it as it is for now. :+1: